### PR TITLE
added QuestKace - Beta

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [x] Ready

## Related Issues
fixes:  demisto/etc#16829

## Description
New integration , related pr: https://github.com/demisto/content/pull/6595

### CODE REVIEW

File: Packs/QuestKace/Integrations/QuestKace/QuestKace.py [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+                 f"Last fetch was on {str(parsed_last_time)}")
+    client.update_token()
+    items: dict = client.tickets_list_request(fetch_shaping, fetch_filter_for_query)
+    items: list = items.get('Tickets', [])
+    incidents, parsed_last_time = parse_incidents(items, fetch_limit, time_format, parsed_last_time)
+    parsed_last_time = parsed_last_time.strftime(time_format)
+    demisto.info(f"Fetching Incident has Finished\n"
+                 f"Fetch limit was {fetch_limit}"
+                 f"Last fetch was on {str(parsed_last_time)}\n"
+                 f"Incidents list is {str(incidents)}")
```
Comment (bakatzir) [inline Packs/QuestKace/Integrations/QuestKace/QuestKace.py:695]: let's rm this line, a bit overloading the logs.

File: Packs/QuestKace/Integrations/QuestKace/QuestKace.py [hunk 2]
```diff
@@ -0,0 +1,907 @@
...
+
+    parsed_last_time = datetime.strptime(new_last_run.get('last_fetch', ''), time_format)
+    filter_after_last_run = f'created gt {parsed_last_time}'
+    fetch_filter_for_query = f'{filter_after_last_run}'
+    if fetch_queue_id:
+        queue_id_str = ';'.join(fetch_queue_id)
+        filter_by_queue_id = f'hd_queue_id in {queue_id_str}'
+        fetch_filter_for_query = f'{fetch_filter_for_query},{filter_by_queue_id}'
+    if fetch_filter:
+        fetch_filter_for_query = f'{fetch_filter_for_query},{fetch_filter}'
```
Comment (bakatzir) [inline Packs/QuestKace/Integrations/QuestKace/QuestKace.py:683]: maybe we can rm one of the vars?
e.g:
rm 676 and
fetch_filter_for_query = f'created gt {parsed_last_time}'

File: Packs/QuestKace/Integrations/QuestKace/QuestKace.py [hunk 3]
```diff
@@ -0,0 +1,907 @@
...
+
+        incident = {
+            'name': item['title'],
+            'occurred': incident_created_time.strftime(time_format),
+            'rawJSON': json.dumps(item)
+        }
+
+        incidents.append(incident)
+        count += 1
+        parsed_last_time = incident_created_time
```
Comment (bakatzir) [inline Packs/QuestKace/Integrations/QuestKace/QuestKace.py:830]: do we have to set it every time? why not just take it from the last item in items outsude of the loop?
also, let's rename it something else throughout the code, to imply that it came from an incident. e.g: `last_incident_time`

File: Packs/QuestKace/Integrations/QuestKace/QuestKace.py [hunk 4]
```diff
@@ -0,0 +1,907 @@
...
+        fetch_filter_for_query = f'{fetch_filter_for_query},{filter_by_queue_id}'
+    if fetch_filter:
+        fetch_filter_for_query = f'{fetch_filter_for_query},{fetch_filter}'
+    demisto.info(f"Fetching Incident has Started,\n"
+                 f"Fetch filter is {fetch_filter_for_query}\n"
+                 f"Last fetch was on {str(parsed_last_time)}")
+    client.update_token()
+    items: dict = client.tickets_list_request(fetch_shaping, fetch_filter_for_query)
+    items: list = items.get('Tickets', [])
+    incidents, parsed_last_time = parse_incidents(items, fetch_limit, time_format, parsed_last_time)
```
Comment (bakatzir) [inline Packs/QuestKace/Integrations/QuestKace/QuestKace.py:690]: have a different name, or else it is misleading to think that this vars have the same value.
```suggestion
    incidents, last_incident_time = parse_incidents(items, fetch_limit, time_format, parsed_last_time)
```

File: Packs/QuestKace/Integrations/QuestKace/QuestKace.py [hunk 5]
```diff
@@ -0,0 +1,907 @@
...
+    parsed_last_time = parsed_last_time.strftime(time_format)
+    demisto.info(f"Fetching Incident has Finished\n"
+                 f"Fetch limit was {fetch_limit}"
```
Comment (bakatzir) [inline Packs/QuestKace/Integrations/QuestKace/QuestKace.py:693]:
```suggestion
                 f"Fetch limit was {fetch_limit}, fetched {len(incidents)} incidents"
```

File: Packs/QuestKace/Integrations/QuestKace/QuestKace.py (hunk 3 above)
Comment (orhovy) [reply Packs/QuestKace/Integrations/QuestKace/QuestKace.py:830]: we do not have to set it every time , The last one of the loop is already converted to a string from date object and I expect during the whole functions date object. So this requires the time of all the fetch function.

**Comment by kirbles19** [comment]
@orhovy Done.

File: Packs/QuestKace/Integrations/QuestKace/README.md [hunk 6]
```diff
@@ -0,0 +1,1253 @@
+This is the Quest KACE integration.
+This integration was integrated and tested with version xx of QuestKace
+
+## Limitations
+Tickets custom fields will not be returned in the **kace-tickets-list** command and not in fetch incidents due to api limitation.
```
Comment (bakatzir) [inline Packs/QuestKace/Integrations/QuestKace/README.md:5]:
```suggestion
Tickets custom fields will not be returned in the **kace-tickets-list** command and not in fetch incidents due to API limitation.
```

File: Packs/QuestKace/Integrations/QuestKace/README.md [hunk 7]
```diff
@@ -0,0 +1,1253 @@
+This is the Quest KACE integration.
+This integration was integrated and tested with version xx of QuestKace
```
Comment (bakatzir) [inline Packs/QuestKace/Integrations/QuestKace/README.md:2]:
```suggestion
This integration was integrated and tested with v10.0.290 of QuestKace
```

File: Packs/QuestKace/TestPlaybooks/QuestKace_test.yml [hunk 8]
```diff
@@ -0,0 +1,522 @@
+id: QuestKace test
+version: -1
+name: QuestKace test
```
Comment (bakatzir) [inline Packs/QuestKace/TestPlaybooks/QuestKace_test.yml:3]:
```suggestion
name: QuestKace test
description: Testing Dell Quest KACE integration.
```

**Pull request merged**